turbine/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -242,29 +242,28 @@ impl StandardBroadcastRun {
                 // Refrain from generating shreds for the slot.
                 return Err(Error::DuplicateSlotBroadcast(bank.slot()));
             }
+
             // Reinitialize state for this slot.
-            let chained_merkle_root = (self.slot == bank.parent_slot())
-                .then_some(self.chained_merkle_root)
-                .ok_or_else(|| {
-                    broadcast_utils::get_chained_merkle_root_from_parent(
-                        bank.slot(),
-                        bank.parent_slot(),
-                        blockstore,
-                    )
-                })
+            self.slot = bank.slot();
+            self.parent = bank.parent_slot();
+            if self.slot != bank.parent_slot() {
```

Comment:
This does not look right. This is effectively checking:
```rust
if bank.slot() != bank.parent_slot() {
```
which is not equivalent to the old code.

Reply:
Fixed.

Thank you for double-checking.
I moved the assignment into the "update state" block and missed the point that `self.slot` was updated %)